### ISSUE TITLE
`export default class implements X` transpilation error

### ISSUE THREAD

**Issue by Electroid** [oven-sh/bun#1789]
@robobun

```ts
export type TestRunner = {
  runTest(path: string): AsyncGenerator<number>;
}

export default class implements TestRunner {
  async* runTest(path: string): AsyncGenerator<number> {
    yield Math.random();
  }
}
```

**Comment by robobun** [comment]
@Electroid here you go!
```tsx
error: Expected "{" but found "TestRunner"
export default class implements TestRunner {
/tmp/bun-KoPngA/index.ts:       ^
/tmp/bun-KoPngA/index.ts:5:33 115
error: Expected identifier but found "{"
                                           {
/tmp/bun-KoPngA/index.ts                   ^
/tmp/bun-KoPngA/index.ts:5:44 126
error: Expected ";" but found "*"
  async* runTest(path: string): AsyncGenerator<number> {
/tmp/bun


error: KoPngA/index.ts:6:8 136
error: Cannot use "yield" outside a generator function
    yield Math.random();
    ^


mp/bun-KoPngA/index.ts:7:5 191
error: Expected ";" but found "Math"
    yield Math.random();
          ^
/tmp/bun-KoPngA/index.ts:7:11 197
error: Unexpected .
    yield Math.random();
              ^
/tmp/bun-KoPngA/index.ts:7:15 201
```
<details>
  <summary>Code</summary>
  
```tsx
export type TestRunner = {
  runTest(path: string): AsyncGenerator<number>;
}

export default class implements TestRunner {
  async* runTest(path: string): AsyncGenerator<number> {
    yield Math.random();
  }
}
```
</details>

_Ran using the latest build of Bun, an all-in-one JavaScript runtime._

**Comment by Electroid** [comment]
Fixed in [88ffdc5](https://github.com/oven-sh/bun/commit/88ffdc5fec9d2f99e891fc6a282c940f08428cfb)
